Demo/src/main/java/com/braintreepayments/demo/PayPalRequestFactory.java

```diff
@@ -18,26 +18,32 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Optional;
 
 public class PayPalRequestFactory {
 
     public static PayPalVaultRequest createPayPalVaultRequest(
         Context context,
         String buyerEmailAddress,
         String buyerPhoneCountryCode,
-        String buyerPhoneNationalNumber
+        String buyerPhoneNationalNumber,
+        String shopperInsightsSessionId
     ) {
 
         PayPalVaultRequest request = new PayPalVaultRequest(true);
 
-        if (!buyerEmailAddress.isEmpty()) {
```

Comment:
Do we need to change all these other params? It seems like it should be unaffected by this PR

Reply:
My understanding is that the original code is potentially a run time crash if the value is null. In the case of the demo app and how its setup, setting the toggles to null didn't have an impact on this flow because the values would always be true.

Reply:
Ah ok in that case do we need to check that it's not null _and_ it's not empty?

Reply:
Yes, great catch!